### PR TITLE
Update the main readme file to be more user-focused

### PR DESCRIPTION
**Pull request by willingc**
- Add more detailed descriptions
- Edit style and heading levels for consistency
- Add file navigation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,59 @@
...
 
 The Python scientific stack, compiled to WebAssembly.
 
-It provides transparent conversion of objects between Javascript and Python.
-When inside a browser, this means Python has full access to the Web APIs.
+[**Try Pyodide and Iodide in your browser**](https://alpha.iodide.io/notebooks/300/)
+
+## What is Pyodide?
+
+**Pyodide** brings the Python runtime to the browser via WebAssembly, along with the Python scientific stack including NumPy, Pandas, Matplotlib, parts of SciPy, and NetworkX.
```
Comment (mdboom) [inline README.md]: maybe add "for a full list [look here](https://github.com/iodide-project/pyodide/tree/master/packages)"

File: README.md [hunk 2]
```diff
@@ -1,30 +1,59 @@
...
+- [Build Pyodide from source](#building-from-source) (this method requires installing prerequistes and using `make`. Primarily for Linux users who want to experiment or contribute back to the project.)
+- [Use a Docker image](#using-docker) (recommended for Windows and macOS users and for Linux users who prefer a Debian-based Docker image on Docker Hub with the dependencies
+already installed)
 
-# Building
+
+### Download a pre-built version
+
+Pre-built versions of Pyodide may be downloaded from
+this repos's [releases page](https://github.com/iodide-project/pyodide/releases/).
```
Comment (mdboom) [inline README.md]: Nit: Maybe say "repository's" (just felt a little weird to abbreviate with the `'s`)...

Comment (willingc) [reply README.md]: Makes sense.

File: README.md (hunk 1 above)
Comment (willingc) [reply README.md]: Great suggestion. Now I know where to find the list too. Thanks for the review.

**Pull request merged**